azure-quantum/azure/quantum/workspace.py

```diff
@@ -374,8 +374,20 @@ def login(self, refresh:bool=False) -> Authentication:
             the user credentials to authenticate with Azure Quantum
         """
 
-        if refresh or (self.credentials is None):
-            msal_wrapper = MsalWrapper(self.subscription_id, refresh=refresh)
-            self.credentials = msal_wrapper.acquire_auth_token()
+        # We only clear the credentials if they are of type BasicTokenAuthentication
+        if refresh and isinstance(self.credentials, BasicTokenAuthentication):
+            self.credentials = None
 
-        return BasicTokenAuthentication(self.credentials)
+        if self.credentials is None:

```

Comment:
Testing if it's an instance of `Authentication` to be consistent with line 389
```suggestion
        if self.credentials is None or (refresh and isinstance(self.credentials, Authentication)):
```

Reply:
The `login()` method has been deprecated and does not do anything anymore.